### ISSUE TITLE
overlay a raster layer over plot_3d 

### ISSUE THREAD

**Issue by aguscamacho** [tylermorganwall/rayshader#191]
Dearlist, I am trying to implement this spectacular visualization tool and the only thing that seems to get in the way is to be able of overlapping raster layers over the  3dplot, How couldthat be possible? I tried add_overlay,but I cannot understand  how to proceed from the error message.

require(raster)
altc=getData('alt', country='MOZ',mask=T)
elevation_matrix = raster_to_matrix(altc)

y=raster(https://www.dropbox.com/s/zhsu204du0rgbjp/Tlim_margins_Mozambique.grd?dl=0)

mat_reduced <- resize_matrix(elevation_matrix, 0.7)

mat_reduced |>
  sphere_shade(texture = create_texture(lightcolor = "white",
                                        shadowcolor = "black",
                                        leftcolor = "grey",
                                        rightcolor = "grey",
                                        centercolor = "grey",
                                        cornercolors = c(NW = "#ffc500",
                                                         NE = "#387642",
                                                         SW = "#d27441",
                                                         SE = "#296176")),
               colorintensity=0.12,
               sunangle = 45) |>

  plot_3d(mat_tx_reduced,
          windowsize = c(800, 600),
          zscale = 200,
          zoom = 0.65,
          phi = 80,
          theta = 0,
          fov = 0,
          shadow = TRUE,
          shadowcolor = "#5da5da",
          water = TRUE,
          watercolor= "lightblue",
          waterlinecolor="lightblue")|>
add_overlay(overlay=y, alphalayer = 0.5)
Error in .local(x, i = i, j = j, ..., value) : unused argument (value)

**Comment by aguscamacho** [comment]
Thank you, but how did you plot without the heightmap argument?
When I try to follow your hint, I get:

The "heightmap" argument is absent, with no default value.

Also, when typing ?plot3d, I get "Internal Server Error".

Regards
Agus

Dr. Agustín Camacho Guerrero. Marie Curie Researcher.
http://www.agustincamacho.com
Departamento de Ecología Evolutiva.
Estación Biológica de Doñana. CSIC.

Avenida Américo Vespucio, núm 26

Isla de la Cartuja. CP: 41092.

Sevilla. España.


*"Unámonos por un mundo mejor, sin injusticia, libre y sin fronteras. Pero
con mucho respeto hacia todas las personas y especies*."

Anderson, Perú, simpatizante de Avaaz.

**Comment by aguscamacho** [comment]
Thanks again, but I am afraid the problem is not a confusion with the name
of the object.
Apparently, I need a hillshade image for plot_3d, but when I added one,
created from the same elevation raster, I got another error:

    plot_3d(heightmap=mat_reduced,hillshade=hill,
          windowsize = c(800, 600),
          zscale = 200,
          zoom = 0.65,
          phi = 80,
          theta = 0,
          fov = 0,
          shadow = TRUE,
          shadowcolor = "#5da5da",
          water = TRUE,
          watercolor= "lightblue",
          waterlinecolor="lightblue")

Error in if (any(hillshade > 1 | hillshade < 0, na.rm = TRUE)) { :
Argument not interpretable as logical
 Warning message:
In any(new("RasterLayer", file = new(".RasterFile", name = "", datanotation
= "LOG1S",  :
  Nothing to summarize if you provide a single RasterLayer; see cellStats


Also, I am unclear about what this means: "You will also have to transform
your raster layer into a RGB 3-layer array first"
I am following this tutorial: https://www.rayshader.com/
Where I do not see that step.
Maybe
       array=as.array(alt)
Would get me such array, but where do I insert it in the function?


Best regards
Agus

Dr. Agustín Camacho Guerrero. Marie Curie Researcher.
http://www.agustincamacho.com
Departamento de Ecología Evolutiva.
Estación Biológica de Doñana. CSIC.

Avenida Américo Vespucio, núm 26

Isla de la Cartuja. CP: 41092.

Sevilla. España.


*"Unámonos por un mundo mejor, sin injusticia, libre y sin fronteras. Pero
con mucho respeto hacia todas las personas y especies*."

Anderson, Perú, simpatizante de Avaaz.
